src/discoverers/rust/default.nix

```diff
@@ -5,13 +5,40 @@
 }: let
   l = lib // builtins;
 
-  discover = {tree}: let
+  discoverCrates = {tree}: let
+    cargoToml = tree.files."Cargo.toml".tomlContent or {};
+
+    subdirCrates =
+      l.flatten
+      (l.mapAttrsToList
+        (dirName: dir: discoverCrates {tree = dir;})
+        (tree.directories or {}));
+  in
+    if cargoToml ? package.name
+    then
+      [
+        {
+          inherit (cargoToml.package) name version;
+          inherit (tree) relPath fullPath;
```

Comment:
No need to put the `fullPath` in the project. you can always re-calculate it via `"${tree.fullPath}/${relPath}"`

Reply:
Oh this isn't actually the project spec. Sorry I was confused. Forget about the previous comment